test-frame-common/src/main/java/io/skodjob/testframe/resources/KubeResourceManager.java

```diff
@@ -526,25 +526,77 @@ private <T extends HasMetadata> void createOrUpdateResource(
     }
 
     /**
+     * Deletes resources with wait asynchronously.
+     *
+     * @param resources     The resources to delete.
+     * @param <T>           The type of the resources.
+     */
+    @SafeVarargs
+    public final <T extends HasMetadata> void deleteResourceWithAsyncWait(T... resources) {
```

Comment:
What does this mean? `deleteResourceWithAsyncWait`. For me, does it mean deleting resources asynchronously and then waiting? In that case, it would no longer be asynchronous, but synchronous (if we just take one element). 

I see that user can specify like 30 items and then run this, which would mean first execute delete() for all 30 items and then sync barrier wait. It's a bit confusing for me.

Reply:
It deletes everything user specifies and then it waits in async matter for all resources to be deleted. Otherwise it would wait synchronously (e.g. one by one if the resource is deleted or not).

Reply:
Oh, I se,e so it's basically parallel wait. When I see `waits` (blocking) and async (non-blocking) together...